openpype/hosts/maya/plugins/publish/extract_playblast.py

```diff
@@ -43,8 +43,17 @@ def _capture(self, preset):
                     json.dumps(preset, indent=4, sort_keys=True)
                 )
             )
-
-        path = capture.capture(log=self.log, **preset)
+        if "textures" in preset["viewport_options"] and (
+            "reloadTextures" in preset["viewport_options"]
+        ):
+            with lib.material_loading_mode():
+                lib.reload_textures()
+                # not supported by `capture`
+                preset["viewport_options"].pop("reloadTextures", None)
+                path = capture.capture(log=self.log, **preset)
+        else:
+            preset["viewport_options"].pop("reloadTextures", None)
+            path = capture.capture(log=self.log, **preset)
```

Comment:
I'd just **always** do the material loading mode thing - irrelevant of whether reload textures was enabled.

Because even if the user _just_ opened their scene I'd still expect them to _want_ to wait for any shader loading of currently visible materials.

Als the `reloadTextures` check only checks for whether a value is present for it. Not whether it's enabled?

I feel like this would be something like:

```python
from contextlib import ExitStack
```

```suggestion
        with ExitStack() as stack:
            if preset["viewport_options"].get("textures"):
                # If capture includes textures then ensure material
                # load mode is set to `immediate` to ensure all
                # textures have loaded when playblast starts
                stack.enter_context(lib.material_loading_mode())
                
            if preset["viewport_options"].get("reloadTextures"):
                # Regenerate all UDIM tiles previews
                lib.reload_textures()

            # not supported by `capture`
            preset["viewport_options"].pop("reloadTextures", None)
            path = capture.capture(log=self.log, **preset)
```

Also, the `reload_textures` function in `lib` now ONLY regenerates the UDIM Tile Previews. As such, I'd refactor that function to have a clearer name and docstring so it refers to what it does, e.g.: `reload_all_udim_tile_previews()`

Reply:
I notice `extract_playblast` has introduced to make sure the code compatible with both Python2 and 3 in terms of contextlib.